### PR TITLE
Add handling of auto search while typing option

### PR DESCRIPTION
**Pull request by fnkhz**
Required for implementing https://github.com/pbek/QOwnNotes/issues/2335
Required by PR: https://github.com/pbek/QOwnNotes/pull/2358

### CODE REVIEW

**Comment by pbek** [comment]
This is a widget library, it shouldn't access the application settings directly!

Plus I guess implementing some kind of "debouncing" (like suggested in https://github.com/pbek/QOwnNotes/issues/1985) might be the better way to go. Any thoughts, @Waqar144?

File: qplaintexteditsearchwidget.cpp [hunk 1]
```diff
@@ -137,6 +138,13 @@ bool QPlainTextEditSearchWidget::eventFilter(QObject *obj, QEvent *event) {
 
 void QPlainTextEditSearchWidget::searchLineEditTextChanged(
     const QString &arg1) {
+
+    QSettings settings;
+    if (arg1.size() != 0 &&
+        settings.value(QStringLiteral("searchMinCharacters"), 0).toInt() > arg1.size()) {
```
Comment (Waqar144) [inline qplaintexteditsearchwidget.cpp:144]: We can't depend on some option from QOwnNotes. This library is independent of QOwnNotes. As such, there is no need for this code here.

**Comment by Waqar144** [comment]
> 

Agreed.

I think its best we just introduce a new option i.e., "start search on pressing enter" or something.

OR

a slightly harder solution i.e., measure if it's taking more than say 200ms and automatically switch off "Search as you type".

**Comment by fnkhz** [comment]
@pbek @Waqar144 Updated to use the suggested "start search on pressing enter" option.

**Comment by fnkhz** [comment]
If debounce is the better option, I could add an option for debounce delay instead, and work with that.

**Comment by pbek** [comment]
> Updated to use the suggested "start search on pressing enter" option.

I can't see any code that suggests that there will be a search "on enter". 😁 

> If debounce is the better option, I could add an option for debounce delay instead, and work with that.

Yes, I still think "debouncing" would be great...

**Comment by fnkhz** [comment]
@pbek It's not "search on pressing enter", it is "auto search on typing .." and it defaults to true.  :smile:

**Comment by pbek** [comment]
I close this in favor of the "de-bouncing2...

**Pull request closed without merging**